docs/registry-docs/copying-images.md

```diff
@@ -1,198 +1,143 @@
 
-## Rancher
+# fetch the content from the carbide secured registry
+# verify the version, location of the key, and the platform/architecture
+hauler store sync --products longhorn=v1.6.0 --key carbide-key.pub --platform linux/amd64
 
-### Cert Manager
+# copy the content from the hauler store to your registry
+hauler store copy --username <redacted> --password <redacted> registry://<registry-url>
+```
+
+## NeuVector
 
 ```bash
-# Log into the registry
+# authenticate into carbide secured registry
 cosign login -u <redacted> -p <redacted> rgcrprod.azurecr.us
 
-# Your target registry (and login if it requires authentication)
-TARGET_REGISTRY=YOUR_REGISTRY_DOMAIN_HERE
-# cosign login -u YOUR_USER -p YOUR_PASSWORD $TARGET_REGISTRY
-
-# Rancher supports v1.7.1 currently. See here for more info:
-# https://ranchermanager.docs.rancher.com/pages-for-subheaders/install-upgrade-on-a-kubernetes-cluster#4-install-cert-manager
-CERT_MANAGER_RELEASE="v1.7.1"
+# download the public key for carbide
+curl -sfl https://github.com/rancherfederal/carbide-releases/releases/download/0.1.1/carbide-key.pub
 
-# Add the cert manager repo (required Helm)
-helm repo add jetstack https://charts.jetstack.io
-helm repo update
+# fetch the content from the carbide secured registry
+# verify the version, location of the key, and the platform/architecture
+hauler store sync --products neuvector=v5.3.0 --key carbide-key.pub --platform linux/amd64
 
-# Grab the list of images and download them (requires docker, grep, sed, and awk)
-for image in $(helm template jetstack/cert-manager --version $CERT_MANAGER_RELEASE | grep 'image:' | sed 's/"//g' | awk '{ print $2 }'); do
-    source_image=$(echo $image | sed 's/quay.io/rgcrprod.azurecr.us/g')
-    dest_image=$(echo $image | sed "s/quay.io/$TARGET_REGISTRY/g")
-    cosign copy $source_image $dest_image
-done
+# copy the content from the hauler store to your registry
+hauler store copy --username <redacted> --password <redacted> registry://<registry-url>
 ```
 
-See [Rancher Manager Configuration](rancher-config.md) for configuring the Cert Manager chart to use these images.
-
-### Rancher
+## Kubewarden
 
 ```bash
-# Log into the registry
+# authenticate into carbide secured registry
 cosign login -u <redacted> -p <redacted> rgcrprod.azurecr.us
 
-# Your target registry (and login if it requires authentication)
-TARGET_REGISTRY=YOUR_REGISTRY_DOMAIN_HERE
-# cosign login -u YOUR_USER -p YOUR_PASSWORD $TARGET_REGISTRY
+# download the public key for carbide
+curl -sfl https://github.com/rancherfederal/carbide-releases/releases/download/0.1.1/carbide-key.pub
 
-# Set the specific release of Rancher you're targeting: https://github.com/rancher/rancher/releases
-RANCHER_RELEASE=v2.7.1
+# fetch the content from the carbide secured registry
+# verify the version, location of the key, and the platform/architecture
+hauler store sync --products kubewarden=v2.0.8 --key carbide-key.pub --platform linux/amd64
```

Comment:
Kubewarden seems to publish an image list for all three components (`crds`, `defaults`, and `controller`) for reach release of the `controller`. Wouldn't it be possible to reference it for the `hauler` manifest?

https://github.com/kubewarden/helm-charts/releases/tag/kubewarden-controller-2.0.8

Reply:
Yeah and that's kinda how we grab things for KW now.  The trouble is, for example, KW v1.10 releases but then the individual components get their own versions and then their helm charts get their own versions.  Plus their charts aren't usually published right away.  Plus plus, v1.10 has many chart versions.

We could change how we usually track things for KW and just go strictly off of the chart version for `controller` instead of KW official version.  I noticed that you used v2.0.8 in your example so you're kinda already thinking that's how it should work.  Do you think that would confuse people since nothing else that we have works this way currently, or am I overthinking the KW version weirdness?

If we can all agree that just tracking KW `controller` chart version is all that we really care about, we could probably build around that.

@atoy3731 @dweomer

Reply:
not as pretty as you probably wanted, but since we're using the chart versions, this is what we get.

`hauler store sync --products kubewarden=kubewarden-controller-2.0.8`

SSF uses the release tag soooo... yep.  :slightly_smiling_face: